R/yaml.R

```diff
@@ -0,0 +1,123 @@
+#' Create a skeleton YAML manifest file
+#'
+#' Tableau extensions require a manifest and the use of a YAML manifest file
+#' in **shinytableau** allows for a separation of concerns when developing an
+#' extension. With the `yaml_skeleton()` function, we can write a YAML manifest
+#' file to disk; it contains all of the required fields for the extension with
+#' example values. Later in the development lifecycle, the developer of a
+#' **shinytableau** extension can use values relevant to the project.
+#'
+#' @param filename The name of the YAML file to create on disk. By default, this
+#'   is `"manifest.yml"`.
+#' @param path An optional path to which the YAML file should be saved (combined
+#'   with `filename`).
+#'
+#' @examples
+#' # Generate a YAML manifest file in the
+#' # root directory of the project
+#' # yaml_skeleton()
+#'
+#' # Generate the YAML file in a subdirectory
+#' # and call the file 'manifest-temp.yml'
+#' # yaml_skeleton(
+#' #   filename = "manifest-temp.yml",
+#' #   path = "app"
+#' # )
+#'
+#' @export
+yaml_skeleton <- function(filename = "manifest.yml",
+                          path = NULL) {
+
+  if (!is.null(path)) {
+    filename <- file.path(path, filename)
+  }
+
+  filename <- fs::path_expand(filename)
+
+  yaml::write_yaml(
+    list(
+      extension_id = "com.example.extensions.name",
+      extension_version = "0.1.0",
+      name = "My Extension",
+      description = NULL,
+      extended_description = NULL,
+      author_name = "Your Name",
+      author_email = NULL,
+      author_organization = NULL,
+      website = "https://example.com",
+      source_location = NULL,
+      icon = list(
+        file = "default_icon.png",
+        package = "shinytableau"
+      ),
+      permissions = "full data",
+      configure = TRUE,
+      min_api_version = "1.4"
+    ),
+    file = filename
+  )
+}
+
+
+#' @export
+tableau_manifest_from_yaml <- function(path = ".") {
```

Comment:
Two things I'd change:

1. I wouldn't use the extension to see if it's a yaml file vs. a directory, instead I'd say, if it's a file, assume it's a yaml file; if it's a directory, assume that it contains the yaml file.
2. I wouldn't just accept any file that happens to be named .yml, instead I think you (Rich) should pick a filename that is required.

Reply:
In case number 2 wasn't clear, I meant you should just decide that the filename must be X (e.g. `manifest.yml`, `tableau_dashboard_extension.yml`, or whatever, just pick one).

Reply:
Thanks for the comments here. I guess to keep this totally unambiguous, the signature for `tableau_manifest_from_yaml` should have separate `file` and `path` (with a `NULL` default) args? For point (2) I'll definitely elect for auto-naming the file `manifest.yml`.